langchain/callbacks/arize_callback.py

```diff
@@ -1,217 +1,314 @@
-from datetime import datetime
-from typing import Any, Dict, List, Optional, Union
+"""
+Callback handlers to log to Arize and Phoenix-compatible file formats.
+"""
+
+import importlib
+import logging
+import os
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from types import ModuleType
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Type,
+    TypeAlias,
+    TypeGuard,
+    Union,
+)
+from uuid import UUID
 
 from langchain.callbacks.base import BaseCallbackHandler
-from langchain.callbacks.utils import import_pandas
-from langchain.schema import AgentAction, AgentFinish, LLMResult
+from langchain.schema.document import Document as LangChainDocument
 
+if TYPE_CHECKING:
+    import pandas as pd
+    from arize.api import Client as ArizeClient
+    from phoenix import Dataset as PhoenixDataset
+    from phoenix import Schema as PhoenixSchema
```

Comment:
We need to avoid having phoenix as a dependency here.

1. Users of Arize are not Phoenix users and vice-versa
2. The Arize client has the right abstractions for logging inferences - phoenix is a server runtime

It's important that these callback managers are fine-grained to specific use-cases so that they can be composed together. For that reason I wouldn't couple phoenix.